File: MyMausam-2.0/packages/types/energy.ts
export interface EnergyGridStatus {
  grid_demand_mw: number;
  solar_generation_mw: number;
  renewable_percentage: number;
  carbon_intensity_gco2_kwh: number;
  optimal_consumption_window: string;
}

export interface ApplianceRecommendation {
  appliance: string;
  recommended_time: string;
  energy_savings_kwh: number;
  co2_reduced_kg: number;
  reason: string;
}

export interface EnergyOptimizationData {
  status: EnergyGridStatus;
  recommendations: ApplianceRecommendation[];
  hourly_demand_forecast: { hour: string; demand_index: number }[];
}

File: MyMausam-2.0/backend/python/app/schemas/energy.py
from typing import List
from pydantic import BaseModel

class ApplianceRecommendationSchema(BaseModel):
    appliance: str
    recommended_time: str
    energy_savings_kwh: float
    co2_reduced_kg: float
    reason: str

class EnergyOptimizationResponse(BaseModel):
    grid_demand_mw: float
    solar_generation_mw: float
    renewable_percentage: float
    carbon_intensity_gco2_kwh: float
    optimal_consumption_window: str
    recommendations: List[ApplianceRecommendationSchema]
